webapps/api/tests/test_authentication.py

```diff
@@ -368,26 +400,39 @@ def test_specify_scopes(self):
         from oauth2.tokenview import OAuth2TokenView
         from view import MetadataView
 
+        time.sleep(180)
         data = {'grant_type': 'password',
                 'username': 'admin',
                 'password': 'admin'}
         request = self.factory.post('/', data=data, HTTP_X_REAL_IP='127.0.0.1')
         response = OAuth2TokenView.as_view()(request)
+        self.assertIn('access_token', json.loads(response.content))
         access_token = json.loads(response.content)['access_token']
+
+        time.sleep(180)
         header = 'Bearer {0}'.format(access_token)
         request = self.factory.get('/', HTTP_X_REAL_IP='127.0.0.1', HTTP_AUTHORIZATION=header)
         response = MetadataView.as_view()(request)
         response_content = json.loads(response.content)
-        self.assertListEqual(response_content['roles'], ['read', 'write', 'manage'])
+        self.assertListEqual(sorted(response_content['roles']), ['manage', 'read', 'write'])
```

Comment:
A set is indeed slightly faster, but it would not validate a possibly duplicate role in the return value.